src/python/pants/backend/python/tasks2/select_interpreter.py

```diff
@@ -11,10 +11,11 @@
 from pex.interpreter import PythonIdentity, PythonInterpreter
 
 from pants.backend.python.interpreter_cache import PythonInterpreterCache
+from pants.backend.python.subsystems.python_setup import PythonSetup
 from pants.backend.python.targets.python_target import PythonTarget
 from pants.base.fingerprint_strategy import DefaultFingerprintHashingMixin, FingerprintStrategy
 from pants.invalidation.cache_manager import VersionedTargetSet
-from pants.python.python_setup import PythonRepos, PythonSetup
+from pants.python.python_repos import PythonRepos
```

Comment:
Not necessarily. PythonRepos are needed both by the python backend and by core pants (in the plugin resolver), so they need to be in core Pants. And PythonRepos represents a clear concept that indeed has nothing specifically to do with the python backend.

PythonSetup, OTOH, is a bit of a hodge-podge of things, but most of them do relate specifically to the Python backend (e.g., how to set up an interpreter cache). So as long as the interpreter cache is in the Python backend and not the core, PythonSetup should be there too.

And indeed, the only thing the core needed from PythonSetup was that cache ttl setting, and that usage was pretty spurious - there's no reason to link "how pants resolves its plugins" with "how pants resolves code it builds".